thunder/core/update_aliases.py

```diff
@@ -0,0 +1,107 @@
+    with tracectx(trace):
+        # !! better to use the meta function?
+        new_alias = TensorProxy(like=alias.proxy, requires_grad=alias.proxy.requires_grad)
+    return new_alias
+
+
+def _get_new_aliases(aliases, trace):
+    return tuple(map(partial(_get_new_alias, trace=trace), aliases))
+
+
+def _is_inplace_op(bsym):
+    return bsym.sym.tags and prims.OpTags.IN_PLACE in bsym.sym.tags
+
+
+def _is_view_creation_op(bsym):
+    import thunder.torch as ltorch
+
+    # !!! check on _syms_that_may_return_views?
+    return bsym.sym in ltorch._syms_returning_views
+
+
+def _involves_viewed_args(bsym, viewed):
+    if bsym.sym.id == prims.PrimIDs.RETURN:
+        return False
+    return any(isinstance(p, TensorProxy) and variableify(p) in viewed for p in bsym.flat_proxy_args)
+
+
+def insert_alias_updates(computation_trace: Trace) -> list[Trace]:
+    swap_map = dict()
+    bsyms = []
+
+    # First pass: identify views, their originals, and operands involved in inplace ops
+    view_groups = []
+    inplace_inputs = set()
+    for bsym in computation_trace.bound_symbols:
+        if _is_inplace_op(bsym) or _is_view_creation_op(bsym):
+            # only interested in the input which is modified by the inplace op
+            in_tensor = list(map(variableify, filter(lambda p: isinstance(p, TensorProxy), bsym.flat_proxy_args)))[0]
+            out_tensors = set(map(variableify, filter(lambda p: isinstance(p, TensorProxy), bsym.flat_proxy_outs)))
+            if _is_inplace_op(bsym):
+                inplace_inputs.add(in_tensor)
+            for group in view_groups:
+                if in_tensor in group:
+                    group.update(out_tensors)
+                    group.add(in_tensor)
+                    break
+            else:
+                view_groups.append(out_tensors.union({in_tensor}))
+
+    # filter out view groups that don't have any tensors involved in inplace ops
+    view_groups = [group for group in view_groups if len(group.intersection(inplace_inputs)) != 0]
+    viewed = set(reduce(set.union, view_groups, set()))
+    encountered = set()
+    for bsym in computation_trace.bound_symbols:
+        if _is_inplace_op(bsym) or _is_view_creation_op(bsym) or _involves_viewed_args(bsym, viewed):
+            in_tensors = map(variableify, filter(lambda p: isinstance(p, TensorProxy), bsym.flat_proxy_args))
+            if _is_inplace_op(bsym):
+                in_tensors = {list(in_tensors)[0]}
+            else:
+                in_tensors = set(in_tensors)
+            out_tensors = set(map(variableify, filter(lambda p: isinstance(p, TensorProxy), bsym.flat_proxy_outs)))
+            encountered.update(in_tensors)
+            group = next((g for g in view_groups if any(g.intersection(in_tensors))), None)
+            if group is None:
+                # this is a view creation with operands that are not involved in any inplace ops
+                bsyms.append(bsym.from_bsym_swap_proxies(swap_map))
```

Comment:
out of curiosity, would bsym get its outputs updated here?

Reply:
No, the output of this as-of-yet untouched bsym would not be present in the swap_map.  The swap_map only has old aliases and the input tensors of in-place operations as keys.

Reply:
understood. How about calling `.from_swap_proxies` with `skip_output=True`?
https://github.com/Lightning-AI/lightning-thunder/blob/92a6adb63ddcf36e3910a55fc5f7a02cec5bee8c/thunder/core/symbol.py#L439-L445